wgpu-core/src/device/resource.rs

```diff
@@ -97,6 +97,7 @@ pub struct Device<A: HalApi> {
     pub(crate) command_allocator: Mutex<Option<CommandAllocator<A>>>,
     //Note: The submission index here corresponds to the last submission that is done.
     pub(crate) active_submission_index: AtomicU64, //SubmissionIndex,
+    // NOTE: if both are needed, the `snatchable_lock` must be acquired before the `fence`
```

Comment:
similar to the note on queue.rs: this should elaborate a little bit on why this is important :)